sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -396,46 +421,59 @@ public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellatio
                                               SendEventOptions options,
                                               CancellationToken cancellationToken = default)
         {
-            options ??= DefaultSendOptions;
+            bool keepTrying = false;
+            int failedAttemptCount = 0;
 
-            Argument.AssertNotNull(events, nameof(events));
-            AssertSinglePartitionReference(options.PartitionId, options.PartitionKey);
+            do
```

Comment:
nit:  Absolutely nothing wrong with how you structured the retries, but could I ask that we adopt the same general structure that the other retry loops take?   We don't need as robust a detection with timeout and delay calculation and similar, but I'd like to try and keep the approach as consistent as we can.

[Example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs#L290)